src/XrdXrootd/XrdXrootdTransit.cc

```diff
@@ -477,11 +464,7 @@ void XrdXrootdTransit::Redrive()
           TRACEP(REQ, "Bridge redrive callback rc="<<rc
                       <<" runStatus="<<runStatus);
         }
-        if (runStatus)
-           {AtomicBeg(runMutex);
-            AtomicZAP(runStatus);
-            AtomicEnd(runMutex);
-           }
+        runStatus.store(0, std::memory_order_release);
```

Comment:
I am really reary about clearing runStatus here. This change was introduced by commit 0588dca
 and I understand what the comments say it wants to fix. However, the logic here does not follow the logic in Process() where runStatus is only cleared if not in runWait. Here it can leave runStatus in an inconsistent state. I still am trying to figure out the implication of that.

Reply:
Happy to make the change but I wanted to point out that this statement, as written, doesn't appear to be true:

> Process() where runStatus is only cleared if not in runWait

From `Process()`:

```
                rc = (Resume ? XrdXrootdProtocol::Process(lp) : Process2());
                if (rc >= 0)
                   {if (runWait) rc = -EINPROGRESS;
                    if (!runDone) return rc;
                    runStatus.store(0, std::memory_order_release);
                   }
```

So `runStatus` is set to 0 if `rc >= 0` and `runDone` is true.  Otherwise, it is not changed.

Now, in this line of code, `runStatus` is unconditionally cleared (regardless of the value of `runDone` or `rc`).  I do not understand the differing logic so I'll defer to you here.